app/interactors/new_document/document_type_selection_interactor.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+class NewDocument::DocumentTypeSelectionInteractor < ApplicationInteractor
+  delegate :params,
+           :user,
+           :redirect_url,
+           :document,
+           :needs_refining,
+           :current_selection,
+           :previous_selection,
+           to: :context
+
+
+  def call
+    find_previous_selection
+    check_for_issues
+    find_current_selection
+    check_for_subtypes
+    find_redirect_url
+
+    if create_document?
```

Comment:
Hmmm. But I don't want to fail. For example, it's valid to not create a document if it's managed elsewhere.

Reply:
I agree. I'm saying we should use the interactor `fail` mechanism in the earlier methods to avoid conditionals later on. The example I linked to should help.

Reply:
I don't understand. There are three possible successful outcomes, only one of which necessitates creating a document. So where would you put the fail? 

I didn't put a guard around `redirect_url` and `needs_refining` because at worst they would only store a nil value in the context, not touch the db.